zboss/CMakeLists.txt

```diff
@@ -1,96 +1,8 @@
-# Those options will be inherited by the ZBOSS target as well as everything
-# that links to it.
-if (CONFIG_ZIGBEE_ROLE_END_DEVICE)
-  target_compile_definitions(zboss-extensions INTERFACE
-    ZB_ED_ROLE
-  )
-endif()
-
-
-# Create a library for Nordic's extensions to ZBOSS stack.
-# The target name is generated based on the path and is not meant to be
-# explicitly linked in other modules.
-# It will be pulled if the module links to the ZBOSS interface library.
-zephyr_library()
-
-# Add source files
-zephyr_library_sources(src/zb_error/zb_error_to_string.c)
-
-# Link with ZBOSS extensions interface library, which is linked to the main
-# ZBOSS interface library.
-zephyr_library_link_libraries(zboss)
-
-# Precompiled libraries -only part.
-# Use the CONFIG_ZBOSS_SOURCES_AVAILABLE Kconfig option is defined in ZBOSS
-# platform module to check if user uses ZOI repositories for building
-# application.
-# If so, the user still may select libraries from nrfxlib to link with
-# verified state of ZBOSS (CONFIG_ZBOSS_LIBRARY_PRECOMPILED symbol).
-if ((NOT DEFINED CONFIG_ZBOSS_SOURCES_AVAILABLE) OR
-    (CONFIG_ZBOSS_LIBRARY_PRECOMPILED))
-
-  nrfxlib_calculate_lib_path(lib_path)
-  set(ZBOSS_LIB_PATH ${CMAKE_CURRENT_SOURCE_DIR}/${lib_path})
-  if(NOT EXISTS ${ZBOSS_LIB_PATH})
-    message(WARNING "This combination of SoC and floating point ABI is not supported by the ZBOSS lib."
-                    "(${ZBOSS_LIB_PATH} doesn't exist.)")
-  endif()
-
-  # Define an interface library for Nordic's ZBOSS stack.
-  # This library contains official ZBOSS headers, platform-specific configuration
-  # and links to Nordic's extensions of ZBOSS API.
-  # This target definition is defined in both - nrfxlib and ZBOSS platform for
-  # NCS due to the lack of control on the order of Zephyr module inclusion.
-  zephyr_interface_library_named(zboss)
-
-  # Extend ZBOSS interface libraries by headers placed in nrfxlib.
-  target_include_directories(zboss INTERFACE
-    include
-    include/zcl
-    include/ha
-    include/se
-    include/osif
-  )
-
-  # Include the static library to the ZBOSS interface definition and select
-  # ZBOSS configuration header file by defining compile-time definition.
-  if (CONFIG_ZIGBEE_ROLE_COORDINATOR OR CONFIG_ZIGBEE_ROLE_ROUTER)
-    target_link_libraries(zboss INTERFACE ${ZBOSS_LIB_PATH}/libzboss.a)
-    target_compile_definitions(zboss INTERFACE LIBZBOSS_CONFIG_FILE="libzboss_config.h")
-  elseif (CONFIG_ZIGBEE_ROLE_END_DEVICE)
-    target_link_libraries(zboss INTERFACE ${ZBOSS_LIB_PATH}/libzboss.ed.a)
-    target_compile_definitions(zboss INTERFACE LIBZBOSS_CONFIG_FILE="libzboss_config.ed.h")
-  else()
-    message( FATAL_ERROR "Unsupported Zigbee role. Exiting." )
-  endif()
-
-  # Link with Nordic's ZBOSS extensions as well as common compile time options
-  # and definitions.
-  target_link_libraries(zboss INTERFACE zboss-extensions)
-endif()
+add_subdirectory_ifdef(CONFIG_ZIGBEE_LIBRARY_PRODUCTION   v3.3)
```

Comment:
`v3.3`, `v3.99`, is this a Zigbee specification version the library implements or the library version itself? In the case of the latter, I'm not sure if it should be encoded in the path as it can change on each release. Perhaps `production`/`development` subdirectories would be a better choice?

Reply:
In case of `v3.3` - It is the ZBOSS major and minor number, so it will change with every update.
The `v3.99` was meant to be "The newest in v3", which used to correspond to the Zigbee R22 PRO specification. In addition, with every major release there may a an API change, so in such case all variants in nrfxlib has to be updated to keep the compatibility with the code in sdk-nrf repository,

I do agree that a different scheme may be better. In my opinion the `production/development` names are a bit too long, but descriptive.

Reply:
I do not have a better idea on the naming convention - decided to go with the `development/production` pair.

Reply:
What do you think about the `stable/unstable` pair?
The point in here is that the `production` is not always `certified`, which may be misleading to some customers.